### PR TITLE
feat(snowflake): Support for OBJECT_INSERT, transpile to DDB

### LINKED ISSUE #3802: Error transpiling OBJECT_CONSTRUCT from Snowflake to DuckDB
Thanks for creating this amazing library! It's very useful for my use-case where I transpile Snowflake queries to DuckDB. I found this bug while migrating some scripts. Not very familiar with the codebase but I can look into it if you provide some insights.

```python
sqlglot.parse_one('select 
     OBJECT_INSERT(OBJECT_INSERT(OBJECT_INSERT(OBJECT_CONSTRUCT(), 'key1', 'value1'), 'key2', 'value2'), 'key3', 'value3')',
 dialect='snowflake').sql(dialect='duckdb')
```

Outputs:

```sql
select Object_insert(Object_insert(Object_insert({}, 'key1', 'value1'), 'key2',
                     'value2'), 'key3', 'value3')     
```

And it fails with the following error:
CleanShot 2024-07-24 at 02 51 18@2x

### PR DESCRIPTION
**Pull request by VaggelisD**
Fixes #3802

Currently, `OBJECT_CONSTRUCT` will transpile to `STRUCT` construction in DuckDB (context https://github.com/tobymao/sqlglot/issues/1699). 

Following this convention, this PR adds support for Snowflake's `OBJECT_INSERT(object, key, value, [update_flag])` which will get transpiled to DuckDB's `STRUCT_INSERT(object, key := value)`. 

Docs
-----------
[Snowflake OBJECT_INSERT](https://docs.snowflake.com/en/sql-reference/functions/object_insert) | [Snowflake OBJECT_CONSTRUCT](https://docs.snowflake.com/en/sql-reference/functions/object_construct) | [DuckDB STRUCT](https://duckdb.org/docs/sql/data_types/struct.html)

### CODE REVIEW

File: sqlglot/dialects/snowflake.py [hunk 1]
```diff
@@ -290,6 +290,7 @@ class Parser(parser.Parser):
             ),
             "NULLIFZERO": _build_if_from_nullifzero,
             "OBJECT_CONSTRUCT": _build_object_construct,
+            "OBJECT_INSERT": exp.ObjectInsert.from_arg_list,
```
Comment (tobymao) [inline sqlglot/dialects/snowflake.py:293]: why is this needed if we have an ObjectInsert expression? shouldn't it auto parse?

Comment (VaggelisD) [reply sqlglot/dialects/snowflake.py]: Right, will remove it.
